components/calendar/src/dangi.rs

```diff
@@ -198,19 +182,23 @@ impl Calendar for Dangi {
     }
 
     fn months_in_year(&self, date: &Self::DateInner) -> u8 {
-        Self::months_for_every_year(date.0 .0.year)
+        date.0.months_in_year_inner()
     }
 
     fn days_in_year(&self, date: &Self::DateInner) -> u16 {
-        Self::days_in_provided_year(date.0 .0.year)
+        date.0.days_in_year_inner()
     }
 
     fn days_in_month(&self, date: &Self::DateInner) -> u8 {
-        Self::month_days(date.0 .0.year, date.0 .0.month)
+        date.0.days_in_month_inner()
```

Comment:
Issue: `Self::month_days` is a trait function that other people are free to call. Isn't it a footgun that `month_days` is slow but `days_in_month` is fast? Should we make the cache accessible to `month_days`?

(similar question for `days_in_provided_year` and `months_for_every_year`)

Reply:
> Issue: `Self::month_days` is a trait function that other people are free to call

This is incorrect, `CalendarArithmetic` is private.

We do have thoughts for performing caching during CalendarArithmetic mutation though